test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/EnvironmentTagHelperTest.cs

```diff
@@ -66,6 +66,97 @@ public void ShowsContentWhenCurrentEnvironmentIsNotSet(string namesAttribute, st
             ShouldShowContent(namesAttribute, environmentName);
         }
 
+
+        [Theory]
+        [InlineData("", "", "", "Development")]
+        [InlineData("", "", "Test", "Development")]
+        [InlineData("Development", "", "", "Development")]
+        [InlineData("", "Development", "", "Development")]
+        [InlineData("", "Development", "Test", "Development")]
+        [InlineData("Development", "", "Test", "Development")]
+        [InlineData("Development", "Development", "", "Development")]
+        [InlineData("Test", "Development", "", "Development")]
+        [InlineData("Development", "Test", "", "Development")]
+        [InlineData("Development", "Development", "Test", "Development")]
+        [InlineData("Development", "Test", "Test", "Development")]
+        [InlineData("Test", "Development", "Test", "Development")]
+        public void ShouldShowContent_IncludeExcludeSpecified(string namesAttribute, string includeAttribute, string excludeAttribute, string environmentName)
+        {
+            // Arrange
+            var content = "content";
+            var context = MakeTagHelperContext(
+                attributes: new TagHelperAttributeList {
+                    { "names", namesAttribute },
+                    { "include", includeAttribute },
+                    { "exclude", excludeAttribute },
+                });
+            var output = MakeTagHelperOutput("environment", childContent: content);
+            var hostingEnvironment = new Mock<IHostingEnvironment>();
+            hostingEnvironment.SetupProperty(h => h.EnvironmentName);
+            hostingEnvironment.Object.EnvironmentName = environmentName;
```

Comment:
Something is busted with mocking if you need both of these lines. The "real" `EnvironmentName` property should never be called.